commerce-service/src/main/java/com/liferay/commerce/internal/order/term/contributor/CommerceRecipientDefinitionTermContributor.java

```diff
@@ -0,0 +1,247 @@
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.ResourceBundle;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Luca Pellizzon
+ */
+@Component(
+	immediate = true,
+	property = {
+		"commerce.definition.term.contributor.key=" + CommerceRecipientDefinitionTermContributor.KEY,
+		"commerce.notification.type.key=" + CommerceOrderConstants.ORDER_NOTIFICATION_AWAITING_SHIPMENT,
+		"commerce.notification.type.key=" + CommerceOrderConstants.ORDER_NOTIFICATION_COMPLETED,
+		"commerce.notification.type.key=" + CommerceOrderConstants.ORDER_NOTIFICATION_PARTIALLY_SHIPPED,
+		"commerce.notification.type.key=" + CommerceOrderConstants.ORDER_NOTIFICATION_PLACED,
+		"commerce.notification.type.key=" + CommerceOrderConstants.ORDER_NOTIFICATION_SHIPPED,
+		"commerce.notification.type.key=" + CommerceOrderConstants.ORDER_NOTIFICATION_TRANSMITTED,
+		"commerce.notification.type.key=" + CommerceSubscriptionNotificationConstants.SUBSCRIPTION_ACTIVATED,
+		"commerce.notification.type.key=" + CommerceSubscriptionNotificationConstants.SUBSCRIPTION_CANCELLED,
+		"commerce.notification.type.key=" + CommerceSubscriptionNotificationConstants.SUBSCRIPTION_RENEWED,
+		"commerce.notification.type.key=" + CommerceSubscriptionNotificationConstants.SUBSCRIPTION_SUSPENDED
+	},
+	service = CommerceDefinitionTermContributor.class
+)
+public class CommerceRecipientDefinitionTermContributor
+	implements CommerceDefinitionTermContributor {
+
+	public static final String KEY =
+		CommerceDefinitionTermConstants.RECIPIENT_DEFINITION_TERMS_CONTRIBUTOR;
+
+	@Override
+	public Map<String, String> getDefinitionTerms(Locale locale) {
+		Map<String, String> map = new HashMap<>();
+
+		List<String> terms = getTerms();
+
+		for (String term : terms) {
+			map.put(term, getLabel(term, locale));
+		}
+
+		return map;
+	}
+
+	@Override
+	public String getFilledTerm(String term, Object object, Locale locale)
+		throws PortalException {
+
+		CommerceOrder commerceOrder = null;
+
+		if (object instanceof CommerceOrder) {
+			commerceOrder = (CommerceOrder)object;
+		}
+		else if (object instanceof CommerceSubscriptionEntry) {
+			CommerceSubscriptionEntry commerceSubscriptionEntry =
+				(CommerceSubscriptionEntry)object;
+
+			CommerceOrderItem commerceOrderItem =
+				_commerceOrderItemLocalService.getCommerceOrderItem(
+					commerceSubscriptionEntry.getCommerceOrderItemId());
+
+			commerceOrder = commerceOrderItem.getCommerceOrder();
+		}
+
+		if (commerceOrder == null) {
+			return term;
+		}
+
+		if (term.equals(_ORDER_CREATOR)) {
```

Comment:
the `term.equals(_ORDER_CREATOR)` block should probably be sorted below the `term.equals(_ACCOUNT_ROLE_ORDER_MANAGER)` block, for the sake of.... alphabetical-ish-ness.